### ISSUE TITLE
Prevent support accounts from deactivating Support Me

### ISSUE THREAD

**Issue by johnbillion** [DrewAPicture/Support-Me#6]
A user with a Support Account role can deactivate the Support Me plugin and then they'll have access to create, edit, delete, and promote users.

The Support Account role shouldn't be able to:
- Deactivate the Support Me plugin.
- Install or delete plugins or themes (activating/deactivating existing ones is fine).
- Access the plugin or theme editors.

**Comment by DrewAPicture** [comment]
I agree about not being able to deactivate the Support Me plugin, but I'm on the fence about not being able to access the editors or install other plugins. Inherently, a user is granting access to a support person to _help them diagnose a problem_. Sometimes that involves installing plugins, perhaps even Debug Bar, Debug Bar Console, or even Query Monitor.

I think at a certain point, if you're trusting some support professional with the keys to the castle in the first place, that you would similarly trust they would be careful not to break anything.

I'll think about the second two.

**Comment by johnbillion** [comment]
That brings into question the reason why SM removes the ability for support accounts to edit users. If it's a courtesy (support users can't easily edit user accounts) as opposed to a security restriction (support users definitely can't edit user accounts) then it makes sense. If it's a security restriction then access to the file editors and adding new plugins needs to be considered.

**Comment by DrewAPicture** [comment]
@johnbillion Following up here, I think the way I'm going to go with this is to add a checkbox to the creation form along the lines of "Allow this account to manage plugins" defaulted to off.

Also disable deactivating Support Me entirely regardless of the setting, and straight up disable the plugin and theme editors as you suggested.

**Comment by zackkatz** [comment]
Reviewed 2026-08-28 — this is the one here with real weight: a support account can deactivate the plugin and thereby escape its own restrictions, which is a privilege-escalation path, not a nice-to-have. It stays open regardless of the maintenance decision — if the plugin is retired, the retirement notice should say this plainly, because installs in the wild carry the issue. Decision tracked in the [Support Me maintenance](https://linear.app/trustedlogin/project/support-me-maintenance-2fd2c206b003) project.
